packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -78,8 +78,11 @@ export class PropertiesManager {
                 if (!newProps[key] && shouldModifyKey(key)) {
                     deltas[key] = oldProps[key];
 
-                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
-                    delete oldProps[key];
+                    const desc = Object.getOwnPropertyDescriptor(oldProps, key);
+                    if (desc?.configurable !== false) {
```

Comment:
i don't really understand how this can happen. i also worry if it happens, we are not being eventually consistent. how do we ensure all clients reach the same state?

Reply:
generally, we can't silently fail when trying write data

Reply:
Yeah, I agree, I don't like this approach either.